Add Officers Teams.

AlbertoPdRF opened the pull request:
After discussing offline with Jeremy about [this comment](https://github.com/thewca/worldcubeassociation.org/pull/3662#issuecomment-450016817), we decided to go ahead with this option.

Implementing this will be useful for #3662 and for record-keeping purposes.

jfly commented on WcaOnRails/app/models/team.rb:
```diff
@@ -8,6 +8,7 @@ class Team < ApplicationRecord
   scope :with_hidden, -> { unscope(where: :hidden) }
 
   scope :official, -> { where(id: Team.all_official.map(&:id)) }
+  scope :officers, -> { where(id: Team.all_officers.map(&:id)) }
```
Is this scope used anywhere?

AlbertoPdRF replied:
Oh, my intention was to use it in [line 38 of the user model](https://github.com/thewca/worldcubeassociation.org/pull/3808/files#diff-572f7c99c29e70c2f7cbb9c219270579R38). I will update it to use it now! However, if you think it's best to not have the scope at all, I can undo the change and remove this line.

jfly commented on WcaOnRails/db/seeds/teams.seeds.rb:
```diff
@@ -12,3 +12,8 @@
 Team.create(friendly_id: 'wst', email: "software@worldcubeassociation.org")
 Team.create(friendly_id: 'banned', email: "disciplinary@worldcubeassociation.org", hidden: true)
 Team.create(friendly_id: 'wcat', email: "competitions@worldcubeassociation.org")
+Team.create(friendly_id: 'chair')
```
I'm not sure if there is any code that assumes that teams have email addresses. I'm ok to leave this as is, but something we should keep in mind in case anything breaks after we deploy this.

AlbertoPdRF replied:
There's no validation for that on the team model, but I don't know if it could be an issue in some place I haven't looked at.

AlbertoPdRF commented on WcaOnRails/db/migrate/20190112130723_add_officers_teams.rb:
```diff
@@ -0,0 +1,25 @@
+    execute <<-SQL
+      INSERT INTO teams (friendly_id, created_at, updated_at) values ('vice_chair', NOW(), NOW());
+    SQL
+  end
```
I used this because it's what we did when creating other teams. I will try with `Team.create` though!

AlbertoPdRF commented:
@jfly Yes, they are the same person, see Section 2 in the [WFC's Motion](https://www.worldcubeassociation.org/documents/motions/10.2017.3%20-%20Financial%20Committee.pdf). I also feel like having the treasurer team is a bit weird, but on the other hand I think being consistent and explicit about the officers roles we have is important too. I don't know what's the best way to go :/

AlbertoPdRF commented:
I've gone ahead and deleted the `treasurer` team, and added a comment explaining that the WCA Treasurer is the WFC Leader before the `officer?` method. I think this is ready to be merged!
